pam/integration-tests/helpers_test.go

```diff
@@ -162,7 +164,11 @@ func saveArtifactsForDebug(t *testing.T, artifacts []string) {
 	// We need to copy the artifacts to another directory, since the test directory will be cleaned up.
 	artifactPath := os.Getenv("AUTHD_TEST_ARTIFACTS_PATH")
 	if artifactPath == "" {
-		artifactPath = filepath.Join(os.TempDir(), "authd-test-artifacts")
+		artifactsFolder := fmt.Sprintf("authd-test-artifacts-%d-%02d-%02dT%02d:%02d:%02d.%d",
+			authdTestSessionTime.Year(), authdTestSessionTime.Month(), authdTestSessionTime.Day(),
+			authdTestSessionTime.Hour(), authdTestSessionTime.Minute(), authdTestSessionTime.Second(),
+			authdTestSessionTime.UnixMilli())
+		artifactPath = filepath.Join(os.TempDir(), artifactsFolder)
```

Comment:
Sure, that's definitely true... But at the same time these are just test artifacts that we wanted in a easily predictable path and only for developers and during their manual tests.

What I wanted to do there was just to be sure that we created a directory that is easy to guess (you know the time) you're running a test and in a location we know about. Indeed we could use `$XDG_RUNTIME_DIR` too, but being only tests-developers code I wouldn't bother much.

In the past everything went to `/tmp/authd-test-artifacts` and that lead to merging results, while the point here was to just save them separately for each test run. If you want to add some extra randomized values we can, but IMHO, given the role of this test code, I wouldn't bother much being perfect as instead we should be in the core features.

Reply:
> but being only tests-developers code I wouldn't bother much.

Sure, it's code only run by developers, but I also don't want to introduce vulnerabilities to the machines I use for development.

> If you want to add some extra randomized values we can

Are you fine with using

```
tmpDir, err := os.MkdirTemp("", "authd-test-artifacts-")
```

then? That still allows to easily find the newest test artifacts directory via `ls -t /tmp/authd-test-artifacts-* | head -n1`.

Reply:
I can use that plus some timings too since I prefer to be able to parse the time for going back too :)

Reply:
Ok this is done now in https://github.com/ubuntu/authd/pull/583/commits/508c00ac6d637300995da69c12a54ad397790c7d indeed it required other changes since the artifact folder should be generated just once for all the test runs, but it should be good enough in this way.

Reply:
thanks!